internal/project/common/docker.go

```diff
@@ -108,6 +108,11 @@ func (docker *Docker) BuildBaseDroneSteps(output drone.StepService) {
 
 // CompileMakefile implements makefile.Compiler.
 func (docker *Docker) CompileMakefile(output *makefile.Output) error {
+	// Only compile if the Dockerfile is used.
```

Comment:
this fixes a bug where `dockerfile` related code was also called for `Pkgfile`, it worked just because we overrode everything in `Pkgfile` steps, though not common options would have created unneeded  steps.